packages/runtime-html/src/templating/render-context.ts

```diff
@@ -442,6 +444,28 @@ export class RenderContext implements IComponentFactory {
     return this;
   }
 
+  private $registerScopeForAuSlot(targetedProjections: RegisteredProjections | null): void {
+    if (targetedProjections === null) { return; }
+    const scope = targetedProjections.scope;
+    const projectionProvider = this.projectionProvider;
+    const instructions = this.compiledDefinition.instructions.flat();
+    let i = 0;
+    while (i < instructions.length) {
+      const instruction = instructions[i++];
+      if (instruction instanceof HydrateElementInstruction) {
+        const slotInfo = instruction.slotInfo ?? null;
+        if (slotInfo !== null) {
```

Comment:
we should have less code & generated code this way, while perf should be better too.
```suggestion
        const slotInfo = instruction.slotInfo;
        if (slotInfo != null) {
```